qdb/ops/ops.go

```diff
@@ -51,8 +51,9 @@ func AddKeyRangeWithChecks(ctx context.Context, qdb qdb.QDB, keyRange *kr.KeyRan
 	}
 
 	for _, v := range existsKrids {
-		if kr.CmpRangesLess(keyRange.LowerBound, v.LowerBound) && kr.CmpRangesLess(v.LowerBound, keyRange.UpperBound) ||
-			kr.CmpRangesLess(keyRange.LowerBound, v.UpperBound) && kr.CmpRangesLess(v.UpperBound, keyRange.UpperBound) {
+		if kr.CmpRangesLessEqual(keyRange.LowerBound, v.LowerBound) && kr.CmpRangesLess(v.LowerBound, keyRange.UpperBound) ||
```

Comment:
Maybe its time to fix https://github.com/pg-sharding/spqr/pull/276/files#diff-6a2b55b44547e95606a78cb12f14a8009366673e342d5e813bcf2f4ff558f1fdR39

For example, qdb.GetKeyRange method is called twice